### ISSUE TITLE
openshift_ca role does not work on containerized installation

### ISSUE THREAD

**Issue by juanwolf** [openshift/openshift-ansible#6973]
#### Description

I was trying to scale up my cluster, adding a new master. The playbooks failed on this tasks: 

```
fatal: [ip-172-31-1-37.eu-west-2.compute.internal]: FAILED! => {
  "msg": "The conditional check 'hostvars[openshift_ca_host].install_result | changed' failed. The error was: |changed expects a dictionary

The error appears to have been in '/home/ubuntu/ansible/openshift-ansible/roles/openshift_ca/tasks/main.yml': line 20, column 3, but maybe elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

- name: Reload generated facts
 ^ here"
}
```

##### Version

Please put the following version information in the code block
indicated below.

* Your ansible version per `ansible --version`

If you're operating from a **git clone**:

* The output of `git describe`

If you're running from playbooks installed via RPM or
`atomic-openshift-utils`

* The output of `rpm -q atomic-openshift-utils openshift-ansible`

Place the output between the code block below:

```
ansible 2.4.2.0
  config file = /home/ubuntu/ansible/openshift-ansible/ansible.cfg
  configured module search path = [u'/home/ubuntu/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]

$ git describe
openshift-ansible-3.7.27-1-13-ga23fea3

```

##### Steps To Reproduce

On an existing cluster 
1. Provision a new master instance
2. Run the `playbooks/byo/openshift-master/scaleup.yml` playbook

##### Expected Results
Describe what you expected to happen.

```
The playbooks running just fine.
```

##### Observed Results
Describe what is actually happening.

```
[ip-172-31-1-37.eu-west-2.compute.internal]: FAILED! => {"msg": "The conditional check 'hostvars[openshift_ca_host].install_resu
lt | changed' failed. The error was: |changed expects a dictionary\n\nThe error appears to have been in '/home/ubuntu/ansible/openshift
-ansible/roles/openshift_ca/tasks/main.yml': line 20, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem
.\n\nThe offending line appears to be:\n\n\n- name: Reload generated facts\n  ^ here\n"}
```

##### Additional Information

Provide any additional information which may help us diagnose the
issue.

* Your operating system and version, ie: RHEL 7.2, Fedora 23 (`$ cat /etc/redhat-release`)
* Your inventory file (especially any non-standard configuration parameters)
* Sample code, etc

```
OS: * centos-atomic-host:centos-atomic-host/7/x86_64/standard
                   Version: 7.1712.1 (2018-01-12 10:25:04)
                    Commit: c6104b6bd8aeabbc19687e91359657e1292c3669e3c686438993e48481af6f2d
              GPGSignature: Valid signature by 64E3E7558572B59A319452AAF17E745691BA8335

```

I inspected the code and I assume it's in the `roles/openshift_ca/tasks/main.yml` file at this part:

```
- name: Install the base package for admin tooling
  package:
    name: "{{ openshift.common.service_type }}{{ openshift_pkg_version | default('') | oo_image_tag_to_rpm_version(include_dash=True) }}"
    state: present
  when: not openshift.common.is_containerized | bool
  register: install_result
  until: install_result | success
  delegate_to: "{{ openshift_ca_host }}"
  run_once: true

- name: Reload generated facts
  openshift_facts:
  when: hostvars[openshift_ca_host].install_result | changed
```
As the install_result variable does not get registred in a containerized deployment, the next task fails. I'll give a try adding the `  when: not openshift.common.is_containerized | bool` statement to the task and update this issue if that fixed it.

**Comment by juanwolf** [comment]
I ran the playbook and this fixed the actual issue. The playbook still failed but that will be for another issue. I'll make a PR in a sec

**Comment by vrutkovs** [comment]
Should be fixed by #6937

**Comment by juanwolf** [comment]
I can see the fix has been applied to the release3.7 branch. I am closing this issue. Thanks again for the quick support.
